### PR TITLE
Implement campaign data tracking.

### PR DESCRIPTION
**Pull request by philosoralphter**

### CODE REVIEW

**Comment by philosoralphter** [comment]
I'll let you take a look at this @jad939933

File: src/com/woopra/tracking/android/WoopraEvent.java [hunk 1]
```diff
@@ -50,6 +73,69 @@ public void setTimestamp(long timestamp) {
...
+	public String getCampaignId() {
+		return campaignProperties.get("campaign_id");
+	}
+
+	public void setCampaignId(String campaignId) {
+		campaignProperties.put("campaign_id", campaignId);
+	}
+
+	public String getCampaignSource() {
+		return campaignProperties.get("campaign_source)";
```
Comment (alexblack) [inline src/com/woopra/tracking/android/WoopraEvent.java]: Typo here

File: src/com/woopra/tracking/android/WoopraEvent.java [hunk 2]
```diff
@@ -22,11 +22,32 @@
...
+	private String campaignName = null;
+	private String campaignId = null;
+	private String campaignSource = null;
+	private String campaignTerm = null;
+	private String campaignMedium = null;
+	private String campaignContent = null;
+
 	private final Map<String, String> properties = new java.util.concurrent.ConcurrentHashMap<String, String>();
 
+	private final Map<String, String> campaignProperties = createCampaignPropsmap()
```
Comment (alexblack) [inline src/com/woopra/tracking/android/WoopraEvent.java]: Semi colon missing :)

**Comment by alexblack** [comment]
I like the direction!  But, intuitively I thought it would it make more sense for these to be on `WoopraVisitor` rather than `WoopraEvent`?

thoughts?

**Comment by eliekhoury** [comment]
@alexblack Campaign properties are stored with events not as visitor properties for many reasons. The most important reason is that a user can come from multiple campaigns. We want to keep track of all the campaigns they came from so we can do better attribution.

Woopra reserved the following event properties which are made available in the user interface for any event without having to update the schema:

- campaign_name
- campaign_source
- campaign_medium
- campaign_content
- campaign_term

**Comment by alexblack** [comment]
@eliekhoury Ok, thanks for the explanation!  

We'd like to be able to pivot our funnels into cohorts by campaign source, does this support that?  The common case for us is that a user found us through a *single* campaign.

If these values are per-event, then yes they are super accurate and precise, and can represent multiple attribution, but, I think our ability to analyze in Woopra will be hampered.

We might consider setting them as visitor props, last in wins, so we can pivot based on them?

**Comment by eliekhoury** [comment]
@alexblack absolutely. You should add your landing page (or whatever event you have that includes the campaign properties) **as a first goal**. For further details, please refer to our docs or reach out to support@woopra.com. Thanks!

**Pull request closed without merging**